department-of-veterans-affairs/va.gov-cms
# VACMS-13639: Fixes content lock operation on Reusable Q&A Content Type

## Description

The root cause of the issue was the selected widget (`Paragraphs(stable)`) for the Related Information field. If no value is provided for this field, the form will be forced to re-render up to five times in an effort to resolve the field.  The content lock module, however WILL NOT render its messages, unlock button, etc. after the first render attempt. Q&A Records that have a Related Information record present will properly render on the first pass and therefore, not subject to this issue.  Changing the widget to `Paragraphs Legacy` allows for proper operation of the content lock module.

Closes #13639 

## Testing done

Performed manual testing on ~20 Q&A records w/o Related Information and ~20 Q&A records with it.  
Confirmed the `This content is now locked by...` system message near top of page, as well as the `Unlock` button near the bottom.
Also, confirmed proper operation -- clicking unlock redirects to Lock Release form.

## Screenshots

### 15637 (Missing Related Information)

![VACMS-13639--20231023--1141](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/607178/1e273cdf-8e98-4949-ad7f-e035f474c8a5)
![VACMS-13639--20231023--1142](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/607178/f6eb2e7c-9298-4e90-93b1-5f463c72215f)

### 47213 (Related Information Populated)
![VACMS-13639--20231023--114230](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/607178/ee0c045a-a170-4419-bf55-399d77a63f0a)
![VACMS-13639--20231023--1143](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/607178/ecce411c-2dc8-4515-95cb-8cdb5e1e1715)


## QA steps

What needs to be checked to prove this works?

Navigating to https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/NID/edit substituting NID with a value below should display similar to the screenshots above:

Q&A w/Related Information:
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/15710/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/15711/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/15713/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/15718/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/15805/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/15806/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/15807/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/15808/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/15809/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/15855/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/15857/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/15858/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/16009/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/16481/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/16583/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/16584/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/16585/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/16586/edit

Q&A w/o Related Information:
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/15712/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/15715/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/15803/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/15804/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/15832/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/15856/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/16482/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/16483/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/20486/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/20488/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/23233/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/23245/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/23246/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/23248/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/23249/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/23250/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/23251/edit 
https://pr15790-g7lgaqxdafmhidbxrj9fq9t1euwnh5tb.ci.cms.va.gov/node/23252/edit


### Select Team for PR review
